### PR TITLE
emscripten types

### PR DESCRIPTION
**Pull request by CedricGuillemet**

### CODE REVIEW

File: recastjs/recast.d.ts [hunk 1]
```diff
@@ -111,4 +113,11 @@ declare module "recast-detour" {
...
-}
\ No newline at end of file
+  export interface RecastDetourModule extends EmscriptenModule {
+  }
+
+  declare const recastdetour_wasm: EmscriptenModuleFactory<RecastDetourModule>
+  export default recastdetour_wasm
+}
+
+export = recastdetour_wasm; // make it a module
```
Comment (RaananW) [inline recastjs/recast.d.ts:123]: not sure this is needed. It will fail under different build paradigms.

File: recastjs/recast.d.ts [hunk 2]
```diff
@@ -111,4 +113,11 @@ declare module "recast-detour" {
     getDefaultQueryExtent(): Vec3;
     getCorners(idx: number): NavPath;
   }
-}
\ No newline at end of file
+  export interface RecastDetourModule extends EmscriptenModule {
```
Comment (RaananW) [inline recastjs/recast.d.ts:116]: I believe this interface should hold the entier implementation (the classes, objects exported, etc').

Comment (RaananW) [reply recastjs/recast.d.ts:116]: Have you tested this declaration in a project? I can do it later

Comment (CedricGuillemet) [reply recastjs/recast.d.ts]: I builds with the test project

**Review by RaananW — APPROVED** [review]

**Pull request merged**